src/librustc_mir/interpret/memory.rs

```diff
@@ -53,6 +54,85 @@ pub struct Memory<'a, 'mir, 'tcx: 'a + 'mir, M: Machine<'mir, 'tcx>> {
     pub tcx: TyCtxtAt<'a, 'tcx, 'tcx>,
 }
 
+impl<'a, 'mir, 'tcx, M> Clone for Memory<'a, 'mir, 'tcx, M>
```

Comment:
Can this be derived?

Reply:
Should be possible after the changes I suggested where the `machine` field of the `EvalContext` becomes part of the state.

Reply:
@bjorn3 Because I chose to refactor the fields of `EvalContext` into a separate struct so that it could have a custom hash impl and be looked up by reference in a hash table. See [my comment](https://github.com/rust-lang/rust/pull/51702#discussion_r197505316) for why this might be overkill. Once you do that, everything that was referring to `ctxt.stack` had to be changed to either `ctxt.stack{,_mut}()` or `ctxt.state.stack`.

Reply:
I did this because explicitly because deriving `Clone` for `Memory` would add a `where M: Clone` bound to the implementation, even though only `where M::MemoryData: Clone` is needed.